crates/compute/src/layer.rs

```diff
@@ -1,18 +1,34 @@
 // Copyright 2025 Irreducible Inc.
 
+use std::ops::Range;
+
+use binius_field::Field;
+use binius_math::ArithExpr;
+use binius_utils::checked_arithmetics::checked_log_2;
+
 use super::{alloc::Error as AllocError, memory::ComputeMemory};
+use crate::memory::SizedSlice;
 
 /// A hardware abstraction layer (HAL) for compute operations.
-pub trait ComputeLayer<F> {
+pub trait ComputeLayer<F: Field> {
```

Comment:
Please consider changing the type trait to `TowerTrait`, such that this interface doesn't have to care about multiple types of `Field`-like traits.

Reply:
Does the trait care? I'm not opposed, but I don't see the immediate need.

Reply:
I'm also wondering how is that going to help with the trait.

Reply:
The `ComputeLayer` trait is defined only with reference to `Field` (no tower there), because it's needed for the `ArithExpr<F>` (otherwise it would bot be bounded at all). Unless I'm missing something.

The `CpuLayer` (binius-internal impl.) is parametrized by a `T: TowerFamily` and them implements `ComputeLayer<T::B128>`.